pkg/controller/nodeipam/ipam/range_allocator.go

```diff
@@ -78,17 +83,24 @@ func NewCIDRRangeAllocator(client clientset.Interface, nodeInformer informers.No
 	klog.V(0).Infof("Sending events to api server.")
 	eventBroadcaster.StartRecordingToSink(&v1core.EventSinkImpl{Interface: client.CoreV1().Events("")})
 
-	set, err := cidrset.NewCIDRSet(clusterCIDR, subNetMaskSize)
-	if err != nil {
-		return nil, err
+	// create a cidrSet for each cidr we operate on
+	// cidrSet are mapped to clusterCIDR by index
+	cidrSets := make([]*cidrset.CidrSet, len(clusterCIDRs))
```

Comment:
A map where the key is an address family seems more intuitive to me, thoughts?

Reply:
So index is to account for multiple cidrs per address family in the future?

Reply:
yes